ArmEmulator/Mem.fs

```diff
@@ -579,53 +594,54 @@ module Mem
                                     else Ok 0u
 
         ///Returns the Ra and Rb values
         let preOrPost pre post (inputRecord: MemInstr) (dP: DataPath<'INS>) x (rABTup: (uint32 * uint32)) =
-            let Ra = fst rABTup
-            let Rb = snd rABTup
+            let rA = fst rABTup
+            let rB = snd rABTup
             match inputRecord.InstructionType with
             | Ok LDR -> match (pre,post) with
                         | (true,true) -> Error "LDRexec-interpretingRecord: pre and post can't both be true"
-                        | (true, _) -> Ok (accessMemoryLocation dP (WA (Rb+x)), Rb+x)
-                        | (_, true) -> Ok (accessMemoryLocation dP (WA (Rb)), Rb+x)
-                        | (false,false) -> Ok (accessMemoryLocation dP (WA (Rb+x)), Rb)
+                        | (true, _) -> Ok (accessMemoryLocation dP (WA (rB+x)), rB+x)
+                        | (_, true) -> Ok (accessMemoryLocation dP (WA (rB)), rB+x)
+                        | (false,false) -> Ok (accessMemoryLocation dP (WA (rB+x)), rB)
             | Ok STR -> match (pre,post) with
                         | (true,true) -> Error "LDRexec-interpretingRecord: pre and post can't both be true"
-                        | (true, _) -> Ok (Ok Ra, Rb+x)
-                        | (_, true) -> Ok (Ok Ra, Rb+x)
-                        | (false,false) -> Ok (Ok Ra, Rb) 
+                        | (true, _) -> Ok (Ok rA, rB+x)
+                        | (_, true) -> Ok (Ok rA, rB+x)
+                        | (false,false) -> Ok (Ok rA, rB) 
+            | Error m -> Error (sprintf "interpretingRecord-preOrPost: Given instruction type not STR of LDR\n%A" m)                    
 
         ///Removes unnecessary levels of Result
         let changeType (v: Result<Result<(Result<uint32,string> * uint32),string>,string>) = 
             match v with
             | Ok x ->   match x with
                         | Ok y -> match y with
                                   | (Ok z, u) -> Ok (z,u)
-                                  | (_, u) -> Error "LDRexec-interpretingRecord: Error accesing memory location"
+                                  | (_, _) -> Error "LDRexec-interpretingRecord: Error accesing memory location"
                         | Error m -> Error m
             | Error m -> Error m
 
         let changedToBytes (inputRecord: MemInstr) (a: Result<(uint32 * uint32),string>) =
             resultDotBindTwoInp makeBytes inputRecord.BytesNotWords a 
 
 
-        changedToBytes inputRecord (changeType (resultDotBindTwoInp (preOrPost inputRecord.PreIndexRb inputRecord.PostIndexRb inputRecord dP) (incrementRbValue inputRecord dP (getOrigVal inputRecord dP)) (getOrigVal inputRecord dP)))
+        changedToBytes inputRecord (changeType (resultDotBindTwoInp (preOrPost inputRecord.PreIndexRb inputRecord.PostIndexRb inputRecord dP) (incrementRbValue inputRecord dP) (getOrigVal inputRecord dP)))
```

Comment:
This is an insanely long line! You can always use indentation to make it run over multiple lines